Upgrade Substrate and Frontier

nazar-pc opened the pull request:
Frontier updates are done in https://github.com/paritytech/frontier/pull/1193, Substrate updates are done in [`subspace-v6`](https://github.com/subspace/substrate/tree/subspace-v6) branch in our fork (rebased due to various conflicts, primarily with block relay).

This upgrades us to latest commit in the old Substrate repo, I'll upgrade us to latest `polkadot-sdk` in a separate PR.

Interesting changes:
* migrations: VersionedRuntimeUpgrade: https://github.com/paritytech/substrate/pull/14311
* EVM + Weight v2 support: https://github.com/paritytech/frontier/pull/1039
  `GasLimitPovSizeRatio`, `weight_limit` and `proof_size_base_cost` were introduced, seem to be parachain-specific and set to defaults.
* Add `pending` support for `eth_getBlockByNumber`: https://github.com/paritytech/frontier/pull/1048
  Introduced `pending_block` RPC runtime API, I used implementation from template in the PR

Other important upstream changes that resulted in downstream modifications:
* https://github.com/paritytech/substrate/pull/14387
* https://github.com/paritytech/substrate/pull/14306
* https://github.com/paritytech/substrate/pull/14437

https://github.com/paritytech/substrate/pull/14709 was promising, but not sufficient, so I opened https://github.com/paritytech/polkadot-sdk/issues/1533

### Code contributor checklist:
* [x] I have read, understood and followed [contributing guide](https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md)

nazar-pc committed:
Remove unnecessary dependency `once_cell`

nazar-pc committed:
Remove unused `parity-db` from `subspace-farmer`

nazar-pc committed:
Remove `jemallocator` dependency from `subspace-farmer`, without `parith-db` is it not really necessary

nazar-pc committed:
Upgrade Substrate to fork with latest changes in old repository and corresponding Frontier fork updates

nazar-pc committed:
Update various non-breaking dependencies

nazar-pc committed:
Remove RocksDB from dependencies pulled by Frontier

nazar-pc committed:
Update parity-db to new upstream release that works on Windows

vedhavyas reviewed (dismissed):
construct_runtime seems much more readable now.

LGTM

nazar-pc committed:
Fix `pallet-subspace` tests

nazar-pc committed:
Fix transaction submission in off-chain context after https://github.com/paritytech/substrate/pull/14387

NingLin-P commented on crates/subspace-transaction-pool/src/lib.rs:
```diff
@@ -417,18 +417,12 @@ where
         client.clone(),
-    ));
-
-    // make transaction pool available for off-chain runtime calls.
-    client
-        .execution_extensions()
-        .register_transaction_pool_factory(OffchainTransactionPoolFactory::new(&pool));
```
The test failed due to tx pool is probably because these are deleted and the equivalents are not added.

vedhavyas committed:
upgrade frontier to latest commit.

We had to introduce Inherent data providers since upstream introduced `pending` tags for Eth rpc calls.
When an rpc call with pending tag is received, we construct pending block by building a new block with extrinsics from tx pool.
So they use inherent data providers to provide inherent data. For now, we just provide timestamp since that is always required.
I have added an TODO to handle any new inherents that needs to be present. Ex runtime upgrade can be added in later PR

nazar-pc committed:
Merge pull request #1958 from subspace/frontier_upgrade

Upgrade frontier deps

nazar-pc committed:
Ignore `test_restart_domain_operator` test on Windows for now

nazar-pc committed:
Merge remote-tracking branch 'origin/main' into upgrade-substrate

# Conflicts:
#	crates/subspace-node/Cargo.toml
#	crates/subspace-node/src/domain/cli.rs
